openstudiocore/src/utilities/filetypes/EpwFile.cpp

```diff
@@ -132,67 +132,119 @@ EpwDataPoint::EpwDataPoint(int year,int month,int day,int hour,int minute,
   setLiquidPrecipitationQuantity(liquidPrecipitationQuantity);
 }
 
-boost::optional<EpwDataPoint> EpwDataPoint::fromEpwString(std::string line)
+boost::optional<EpwDataPoint> EpwDataPoint::fromEpwString(const std::string &line)
+{
+  std::vector<std::string> list;
+  boost::split(list, line, boost::is_any_of(","));
+  return fromEpwStrings(list);
+}
+
+boost::optional<EpwDataPoint> EpwDataPoint::fromEpwStrings(const std::vector<std::string> &list, bool pedantic)
 {
   EpwDataPoint pt;
-  QStringList list = QString().fromStdString(line).split(',');
-  // Require 35 items in the list
-  if(list.size() < 35) {
-    // JWD: Should this just use the entries that are there and fill in the rest as unavailable?
-    LOG_FREE(Error,"openstudio.EpwFile","Expected 35 fields in EPW data, got " << list.size());
-    return boost::optional<EpwDataPoint>();
-  } else if(list.size() > 35) {
-    LOG_FREE(Error,"openstudio.EpwFile","Expected 35 fields in EPW data, got " << list.size() << ", additional data will be ignored");
-    return boost::optional<EpwDataPoint>();
+  // Expect 35 items in the list
+  if (list.size() < 35) {
+    if (pedantic) {
+      LOG_FREE(Error, "openstudio.EpwFile", "Expected 35 fields in EPW data, got " << list.size());
+      return boost::none;
+    } else {
+      LOG_FREE(Warn, "openstudio.EpwFile", "Expected 35 fields in EPW data, got " << list.size() << ", remaining fields will not be available");

```

Comment:
@jasondegraw won't this cause an access violation down below when you try to access the fields if list is < 35?